GitUI/UserControls/WarningHeader.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using GitCommands;
+using GitUI.CommandsDialogs;
+using Microsoft.VisualStudio.Threading;
+
+namespace GitUI.UserControls
+{
+    public partial class WarningHeader : UserControl
+    {
+        private enum State
+        {
+            None,
+            Bisect,
+            Rebase,
+            Merge,
+            Patch
+        }
+
+        private State _state = State.None;
+        private bool _hasConflicts = false;
+
+        public event EventHandler BisectEvent;
+
+        public WarningHeader()
+        {
+            InitializeComponent();
+        }
+
+        public async Task RefreshAsync(GitCommands.GitModule module)
```

Comment:
We mustn't pass `GitModule` as a parameter as it causes all sorts of dramas, including seriously inhibiting tests.
Instead please inherit the control from `GitModuleControl` and you will have access to the current instance of `GitModule`.